src/components/registration-form.js

```diff
@@ -1,148 +1,285 @@
+  const handleEnterCodeFinish = async ({ verificationCode }) => {
+    try {
+      setIsLoading(true);
+      const { user } = await userApiClient.completeRegistration({
+        userId: createdUser._id,
+        verificationCode: verificationCode.trim()
+      });
+      setUser(user);
+      setCreatedUser(user);
+      setCurrentStepKey(STEP.completed);
+      setSteps(prev => setStepSuccess(prev, STEP.enterCode));
+    } catch (error) {
+      if (error.status === HTTP_STATUS.notFound) {
+        setInvalidVerificationCodes(prevState => [...prevState, verificationCode]);
+        setTimeout(() => enterCodeForm.validateFields(['verificationCode'], { force: true }), 0);
+      } else {
+        handleApiError({ error, logger, t });
+      }
+      setSteps(prev => setStepError(prev, STEP.enterCode));
+    } finally {
+      setIsLoading(false);
+    }
   };
 
-  const handleRegisterButtonClick = () => {
-    form.submit();
+  const handleGotoDashboardClick = () => {
+    window.location = routes.getDashboardUrl();
   };
 
-  const agreementValidationRules = [
-    {
-      message: t('confirmTerms'),
-      validator: (_, value) => value ? Promise.resolve() : Promise.reject(new Error(t('confirmTerms')))
-    }
-  ];
-
-  const termsPage = settings.termsPage?.[uiLanguage];
-
-  const registrationForm = (
-    <div className="RegistrationForm-form">
-      <Form
-        form={form}
-        layout="vertical"
-        onFinish={handleFinish}
-        validateTrigger="onSubmit"
-        scrollToFirstError
-        >
-        <EmailFormItem name="email" emailsInUse={forbiddenEmails} />
-        <PasswordFormItem name="password" />
-        <PasswordConfirmationFormItem name="confirm" passwordFormItemName="password" />
-        <DisplayNameFormItem name="displayName" />
-        <FormItem name="agreement" valuePropName="checked" rules={agreementValidationRules}>
-          <Checkbox className="RegistrationForm-termsAndConditionsCheckbox">
-            <Trans
-              t={t}
-              i18nKey="termsAndConditionsConfirmation"
-              components={[
-                <a
-                  key="terms-link"
-                  title={termsPage?.linkTitle || null}
-                  href={termsPage?.documentId ? routes.getDocUrl({ id: termsPage.documentId }) : '#'}
-                  />
-              ]}
-              />
-          </Checkbox>
-        </FormItem>
-      </Form>
-      <div className="RegistrationForm-registerButton">
-        <Button type="primary" size="large" onClick={handleRegisterButtonClick} block>
-          {t('register')}
-        </Button>
+  const renderEnterDataForm = () => {
+    const termsPage = settings.termsPage?.[uiLanguage];
+    const agreementValidationRules = [
+      {
+        message: t('confirmTerms'),
+        validator: (_, value) => value ? Promise.resolve() : Promise.reject(new Error(t('confirmTerms')))
+      }
+    ];
+    return (
+      <div className="RegistrationForm-form">
```

Comment:
Small thing and the code is old, but we could drop the `-form` in the class name, as it's the outer-most class in this component.